### PR TITLE
feat: add contributing guide

### PR DESCRIPTION
**Pull request by SputNikPlop**
**This is a work in progress and will seek feedback do not merge**

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

## Problem/Feature

The project does not have a contributing guide that helps people getting started adding their own contributions to the project.

## Description of Changes: 

- Add a contribution guide.

## Checklist

- [x] I have self-reviewed the changes being requested
- [x] I have updated the documentation (if applicable)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Contributing
+
+Thanks for helping make the Twitch CLI better!
```
Comment (mimatache) [inline CONTRIBUTING.md:3]: Shouldn't it be `go-blueprint` instead on `Twitch CLI`?

Comment (Melkeydev) [reply CONTRIBUTING.md:3]: @SputNikPlop Yeah this should say go-blueprint

**Review by Melkeydev — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Contributing
+
+Thanks for helping make the go-blueprint better! 
+- [Contributing](#contributing)
+  - [Design Principles](#design-principles)
+  - [Report an Issue](#report-an-issue)
+  - [Contributing Code with Pull Requests](#contributing-code-with-pull-requests)
+    - [Requirements](#requirements)
```
Comment (mikelerch) [inline CONTRIBUTING.md:8]: Indention isn't consistent

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  - [Licensing](#licensing)
+
+## Design Principles
+
+Contributions to the go-blueprint should align with the project’s design principles:
```
Comment (mikelerch) [inline CONTRIBUTING.md:13]: Grammatical error, should remove "the"

**Review by Melkeydev — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Contributing
+
+Thanks for helping make the go-blueprint better!
```
Comment (mikelerch) [inline CONTRIBUTING.md:3]: Would remove "the"

**Pull request merged**